include/zephyr/data/json.h

```diff
@@ -137,6 +138,9 @@ typedef int (*json_append_bytes_t)(const char *bytes, size_t len,
 				 __alignof__(type) == 2 ? 1 : \
 				 __alignof__(type) == 4 ? 2 : 3)
 
+/* Helper to get the size of a member within a struct. */
+#define Z_MEMBER_SIZE(type, member) sizeof(((type *)0)->member)
```

Comment:
Given how often I needed this in other zephyr projects I'd say it's finally time to add this to `include/zephyr/sys/util.h` with the name `SIZEOF_FIELD` (that's what the linux kernel calls this, just lowercase).